components/drops/useDrops.ts

```diff
@@ -48,28 +49,33 @@ const DROP_LIST_ORDER = [
 
 const ONE_DAYH_IN_MS = 24 * 60 * 60 * 1000
 
-export function useDrops(query?: GetDropsQuery) {
+export function useDrops(query?: GetDropsQuery, config?: { filterOutMinted?: boolean }) {
   const drops = ref<Drop[]>([])
   const dropsList = ref<DropItem[]>([])
+  const { filterOutMinted } = config || {}
   const count = computed(() => dropsList.value.length)
   const loaded = ref(false)
 
   onBeforeMount(async () => {
     dropsList.value = await getDrops(query)
 
     const formattedDrops = await Promise.all(
-      dropsList.value.map(async (drop) => getFormattedDropItem(drop, drop)),
+      dropsList.value.map(async drop => getFormattedDropItem(drop, drop)),
     )
 
     drops.value = formattedDrops
 
+    if (filterOutMinted) {
+      drops.value = drops.value.filter(drop => !drop.isMintedOut)
+    }
```

Comment:
nitpick , this avoids updating the ref twice

``````
  let formattedDrops = await Promise.all(
      dropsList.value.map(async drop => getFormattedDropItem(drop, drop)),
    )

    if (filterOutMinted) {
      formattedDrops = formattedDrops.filter(drop => !drop.isMintedOut)
    }

    drops.value = formattedDrops
``````

or maybe use a `then` and this way you don't have to use `let` and override

Reply:
updated